docs/usage/dns-autoscaling.md

```diff
@@ -0,0 +1,63 @@
+
+- Cloud provider limits for DNS lookups.
+- Unreliable UDP connections that forces a period of timeout in case packets are dropped.
+- Unnecessary node hopping since CoreDNS is not deployed on all nodes, and as a result DNS queries end-up traversing multiple nodes before reaching the destination server.
+- Inefficient load-balancing of services (e.g., round-robin might not be enough when using IPTables mode).
+- Overload of the CoreDNS replicas as the maximum amount of replicas is fixed.
+- and more ...
+
+As an alternative with extended configuration options, Gardener provides cluster-proportional autoscaling of CoreDNS. This guide focuses on the configuration of cluster-proportional autoscaling of CoreDNS and its advantages/disadvantages compared to the horizontal
+autoscaling.
+Please note that there is also the option to use a [node-local DNS cache](node-local-dns.md), which helps mitigate potential DNS bottlenecks, but is not discussed in detail in this guide.
+
+## Configuring cluster-proportional DNS Autoscaling
+
+All that needs to be done to enable the usage of cluster-proportional autoscaling of CoreDNS is to set the corresponding option (`spec.systemComponents.coreDNS.autoscaling.mode`) in the `Shoot` resource to `cluster-proportional`:
+
+```yaml
+...
+spec:
+  ...
+  systemComponents:
+    coreDNS:
+      autoscaling:
+        mode: cluster-proportional
+...
+```
+
+To switch back to horizontal DNS autoscaling you can set the `spec.systemComponents.coreDNS.autoscaling.mode` to `horizontal` (or remove the `coreDNS` section).
+
+Once the cluster-proportional autoscaling of CoreDNS has been enabled and the Shoot cluster has been reconciled afterwards, a ConfigMap called `kube-dns-autoscaler` will be created in the `kube-system` namespace with the default settings. The content will be similar to the following:
+
+```yaml
+linear: '{"coresPerReplica":256,"min":2,"nodesPerReplica":16}'
+```
+
+It is possible to adapt the ConfigMap according to your needs in case the defaults do not work as desired. The number of CoreDNS replicas is calculated according to the following formula:
+
+```
+replicas = max( ceil( cores × 1 / coresPerReplica ) , ceil( nodes × 1 / nodesPerReplica ) )
+```
+
+Depending on your needs, you can adjust `coresPerReplica` or `nodesPerReplica`, but it is also possible to override `min` if required.
+
+## Trade-offs of horizontal and cluster-proportional DNS Autoscaling
+
+The horizontal autoscaling of CoreDNS as implemented by Gardener is fully managed, i.e. you do not need to perform any configuration changes. It scales according to the cpu usage of CoreDNS replicas meaning that it will create new replicas if the existing ones are under heavy load. However, it is limited to scale between 2 and 5 instances.
```

Comment:
```suggestion
The horizontal autoscaling of CoreDNS as implemented by Gardener is fully managed, i.e. you do not need to perform any configuration changes. It scales according to the CPU usage of CoreDNS replicas meaning that it will create new replicas if the existing ones are under heavy load. However, it is limited to scale between 2 and 5 instances.
```

Reply:
As a reader, I would wonder why it is limited to 5 instances while I can tweak the settings for `cluster-proportional` scaling myself. Why can't I tweak this maximum number?

Reply:
The existing autoscaling solution simply does not allow users to configure the maximum amount of instances. There is no technical reason (I am aware of) for this though.
Do you suggest to make the existing solution more configurable as part of this change request or would you rather like to adapt the documentation?

Reply:
> would you rather like to adapt the documentation?

This. While reading it this questions arose in my mind, so maybe we can answer it upfront if others are wondering similarly.

Reply:
Ok, I will try, but frankly I do not understand why it is not configurable.

Reply:
Fixed the casing of `CPU` and adapted the wording of the scale limits of the horizontal autoscaling solution.
Feedback is welcome.